lib/sweet_xml.ex

```diff
@@ -714,7 +714,7 @@ defmodule SweetXml do
   ```
   For more details, see `parse/2`.
   """
-  @spec xmap(parent :: (doc | xmlElement), mapping :: specs, options :: (boolean | map)) :: (map | keyword)
+  @spec xmap(parent :: (doc | xmlElement), mapping :: specs, options :: (boolean | map)) :: (map | keyword | nil)
```

Comment:
Would you mind adding `nil` as an input option as well?

```suggestion
  @spec xmap(parent :: (doc | xmlElement | nil), mapping :: specs, options :: (boolean | map)) :: (map | keyword | nil)
```